### PR TITLE
Upgrade to Python 3.10 and Discord.py 2.2

### PR DESCRIPTION
**Pull request by mvanbaalen**
This is in preparation for slash commands. I upgraded to the newest version of Python since we were past EOL, and upgraded every other package along with it, including discord which has had major updates.

I fixed two bugs (!! triggering the bot and Sunday eamuse being wrong) while I was in there.

I tested every operation the way I normally do except for Youtube. Image search works fine, but I am over my Youtube quota and don't feel like fixing it.

**I did not check that the new dockerfile is correct.** I looked up official dockerfiles and believe I chose one that correctly resolves, but I don't know for sure.

### CODE REVIEW

File: bots/honkbot.py [hunk 1]
```diff
@@ -73,7 +73,8 @@ async def on_message(self, message):
     async def on_command_error(self, ctx, error):
         self.logger.error(error)
         if isinstance(error, commands.CommandNotFound):
-            await ctx.send("Use !help to see a list of commands")
+            if ctx.invoked_with.isalnum():
+                await ctx.send(f"{ctx.invoked_with} is not a command. Use !help to see a list of commands")
```
Comment (mvanbaalen) [inline bots/honkbot.py:77]: My assumption is that we won't be implementing any commands with punctuation, so this should catch stuff like !!! that people normally say and shouldn't trigger the bot.

File: bots/honkbot.py [hunk 2]
```diff
@@ -220,7 +221,7 @@ async def eamuse(self, ctx):
                                + f"{self.get_display_time('daily')} daily. "
                                + f"{self.messages['website_down']}**")
 
-            if datetime.datetime.today().weekday() in range(0, 4):
+            if datetime.datetime.today().weekday() in [6, 0, 1, 2, 3]:
```
Comment (mvanbaalen) [inline bots/honkbot.py:224]: Used to be "on Monday through Friday" but now "on Sunday through Thursday"

File: requirements.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
+discord.py==2.2.2
 python-dotenv>=0.9.1
-pytz==2018.5
-requests==2.20.0
-beautifulsoup4==4.6.3
-psycopg2==2.8.4
+pytz==2022.7.1
+requests==2.28.2
+beautifulsoup4==4.12.0
+psycopg2-binary==2.9.5
```
Comment (mvanbaalen) [inline requirements.txt:6]: In my personal projects and local environment I always have to use -binary, and that was true here. I have never gotten the non-binary to work. Hopefully this isn't a problem, but I believe this is the recommended way anyway.

File: startup.py [hunk 4]
```diff
@@ -19,12 +27,11 @@
     speedrun_api_key = os.getenv("SPEEDRUN_API_KEY")
     google_api_key = os.getenv("GOOGLE_API_KEY")
 
-    bot = Bot(command_prefix='!')
+    intents = Intents(messages=True, message_content=True, guilds=True)
```
Comment (mvanbaalen) [inline startup.py:30]: These are unbelievably important and there is a massive list of them to keep in mind whenever you add new features. I was tearing my hair out for a while not realizing that you can't do anything with roles if you don't have guilds.

Anyway, "messages" lets us react to messages being sent, "message_content" lets us read messages (to get the commands out) and "guilds" lets us get role access.

File: startup.py [hunk 5]
```diff
@@ -19,12 +27,11 @@
...
+    discord_bot = Bot(command_prefix='!', intents=intents)
 
-    honkbot = Honkbot(logger, speedrun_api_key, google_api_key, bot=bot)
-    bot.add_cog(honkbot)
-    bot.add_cog(Remybot())
-    bot.add_cog(Smxbot())
-    bot.add_cog(EamuseRivals())
+    honkbot = Honkbot(logger, speedrun_api_key, google_api_key, bot=discord_bot)
+    bot_cogs = [honkbot, Remybot(), Smxbot(), EamuseRivals()]
+    asyncio.run(add_cogs(discord_bot, bot_cogs))
```
Comment (mvanbaalen) [inline startup.py:35]: Adding cogs is now asynchronous.

**Review by tylernap — APPROVED** [review]

**Pull request merged**